frontend/app/assets/javascripts/spree/frontend/checkout/coupon-code.js

```diff
@@ -30,7 +30,11 @@ Spree.onCouponCodeApply = function(e) {
     var handler;
     // handler = JSON.parse(xhr.responseText)
     handler = xhr.responseJSON;
-    couponStatus.addClass(errorClass).html(handler["error"]);
+    const errorMessages = $.map(
+      handler["errors"],
+      function (err) { return err.error }
+    )
+    couponStatus.addClass(errorClass).html(errorMessages.join('<br>'));
```

Comment:
Do you have a screenshot of how it looks? Doesn't need to be good, just wanted to be sure it doesn't break the page layout.

Reply:
Saw the screenshot in the PR description. All good here!